### PR TITLE
Require Admin to restart Oauth2Proxy

### PR DESCRIPTION
**Pull request by Richard87**

### CODE REVIEW

File: api/kubequery/radixapplication.go [hunk 1]
```diff
@@ -3,14 +3,27 @@ package kubequery
...
 // GetRadixApplication returns the RadixApplication for the specified application name.
 func GetRadixApplication(ctx context.Context, client radixclient.Interface, appName string) (*radixv1.RadixApplication, error) {
 	ns := operatorUtils.GetAppNamespace(appName)
 	return client.RadixV1().RadixApplications(ns).Get(ctx, appName, metav1.GetOptions{})
 }
+
+func IsRadixApplicationAdmin(ctx context.Context, kubeClient kubernetes.Interface, appName string) (bool, error) {
+	return access.HasAccess(ctx, kubeClient, &authorizationapi.ResourceAttributes{
+		Verb:     "patch",
+		Group:    "",
```
Comment (satr) [inline api/kubequery/radixapplication.go:24]: Group:    "radix.equinor.com",

File: api/kubequery/radixapplication.go [hunk 2]
```diff
@@ -3,14 +3,27 @@ package kubequery
...
+		Resource: radixv1.KindRadixRegistration,
```
Comment (satr) [inline api/kubequery/radixapplication.go:25]:
```suggestion
		Resource: "radixregistrations",
```

File: api/kubequery/radixapplication.go (hunk 1 above)
Comment (satr) [inline api/kubequery/radixapplication.go:24]:
```suggestion
		Group: "radix.equinor.com",
```

File: api/kubequery/radixapplication.go (hunk 2 above)
Comment (Richard87) [reply api/kubequery/radixapplication.go]: Fixed

File: api/kubequery/radixapplication.go (hunk 1 above)
Comment (Richard87) [reply api/kubequery/radixapplication.go]: Fixed

File: api/environments/component_handler.go [hunk 3]
```diff
@@ -108,6 +108,11 @@ func (eh EnvironmentHandler) RestartComponent(ctx context.Context, appName, envN
 // RestartComponentAuxiliaryResource Restarts a component's auxiliary resource
 func (eh EnvironmentHandler) RestartComponentAuxiliaryResource(ctx context.Context, appName, envName, componentName, auxType string) error {
 	log.Ctx(ctx).Info().Msgf("Restarting auxiliary resource %s for component %s, %s", auxType, componentName, appName)
+	if isAdmin, err := kubequery.IsRadixApplicationAdmin(ctx, eh.accounts.UserAccount.Client, appName); err != nil {
+		return err
+	} else if !isAdmin {
```
Comment (satr) [inline api/environments/component_handler.go:113]: Not needed else

**Review by satr — APPROVED** [review]

**Pull request merged**